packages/react-badge/src/components/CounterBadge/useCounterBadge.ts

```diff
@@ -1,41 +1,23 @@
 import * as React from 'react';
-import { makeMergePropsCompat } from '@fluentui/react-utilities';
 import { useBadge } from '../Badge/index';
 import type { CounterBadgeProps, CounterBadgeState } from './CounterBadge.types';
-import type { BadgeProps } from '../Badge/index';
-
-/**
- * Consts listing which props are shorthand props.
- */
-export const counterBadgeShorthandPropsCompat = ['icon'] as const;
-
-// eslint-disable-next-line deprecation/deprecation
-const mergeProps = makeMergePropsCompat<CounterBadgeState>({ deepMerge: counterBadgeShorthandPropsCompat });
+import { BadgeState } from '../Badge/index';
 
 /**
  * Returns the props and state required to render the component
  */
-export const useCounterBadge = (
-  props: CounterBadgeProps,
-  ref: React.Ref<HTMLElement>,
-  defaultProps?: CounterBadgeProps,
-): CounterBadgeState => {
-  const state = useBadge(
-    props,
-    ref,
-    mergeProps(
-      {
-        showZero: false,
-        overflowCount: 99,
-        count: 0,
-        dot: false,
-      },
-      defaultProps,
-    ) as BadgeProps,
-  ) as CounterBadgeState;
+export const useCounterBadge = (props: CounterBadgeProps, ref: React.Ref<HTMLElement>): CounterBadgeState => {
+  const state: CounterBadgeState = {
+    ...(useBadge(props, ref) as Omit<BadgeState, 'shape' | 'appearance'> &
+      Pick<CounterBadgeState, 'shape' | 'appearance'>),
```

Comment:
This cast is unfortunate, and potentially unsafe if `useBadge` is changed in the future to return a different default value for `shape` or `appearance`. 

I think it'll work without the cast if you provide values for `shape` and `appearance`. It also doesn't risk an invalid value for one of those props.
```suggestion
    ...useBadge(props, ref),
    shape: props.shape ?? 'circular',
    appearance: props.appearance ?? 'filled',
```

Reply:
yeah that's a good call out, thanks 👍